pkg/backend/openshiftcluster.go

```diff
@@ -292,6 +292,12 @@ func (ocb *openShiftClusterBackend) endLease(ctx context.Context, log *logrus.En
 	var adminUpdateError *string
 	var failedProvisioningState api.ProvisioningState
 	initialProvisioningState := doc.OpenShiftCluster.Properties.ProvisioningState
+	var err error
+
+	defer func() {
+		recover.Panic(log)
+		_, err = ocb.dbOpenShiftClusters.EndLease(ctx, doc.Key, provisioningState, failedProvisioningState, adminUpdateError)
```

Comment:
I don't think this works as expected - `err` is swallowed in the `defer` here.

Here's a playground to show what I mean: https://go.dev/play/p/3Efb6cUNo8U

Reply:
found the fix - use a named return result, in the playground, changing the function header to this works:

func sevenIsTooScary() (err error)

in the `defer`, we are then modifying the named `err` value.

tl;dr - remove `var err error` and change the `endLease` header to return `(err error)` instead of `error`.

Reply:
@kimorris27 
This might help
https://news.ycombinator.com/item?id=14668955
> Yep, and one cannot simulate that with local variables. In Go "return v" copies v into the return location before calling the deferred code. If that location is not named, the deferred function has no way to change it

Reply:
Yeah, that does help, thank you! I learned some new things about Go today.